Arelle/Arelle
# Fix long Updater module

#### Reason for change
#247 

#### Description of change
* For the Updater module:
	* Makes it work again on mac and windows
	* Adds warning messages when something goes wrong
	* Adds type hints
	* Adds test coverage

#### Steps to Test
* load the GUI (`python arelleGUI.pyw`)
* use the check for updates feature in the help menu
* experiment with changing the date in `arelle/Version.version` to be:
	* older than the latest update
	* newer than the latest update
	* the same as the latest update

**review**:
@Arelle/arelle
